File: code/generate_strategies/IR(pseudocode)/java/java/function_seaborn_set_theme.java
package com.example;

import org.jfree.chart.ChartFactory;
import org.jfree.chart.JFreeChart;
import org.jfree.chart.StandardChartTheme;
import org.jfree.chart.renderer.category.StandardBarPainter;
import java.awt.*;
import java.util.Arrays;
import java.util.List;

public class SeabornThemeSetter {
    public String set_theme(String style, String context, String palette) {
        class SeabornThemeSetter {
            private List<String> valid_styles = Arrays.asList("white", "dark", "whitegrid", "darkgrid", "ticks");
            private List<String> valid_contexts = Arrays.asList("paper", "notebook", "talk", "poster");
            private List<String> valid_palettes = Arrays.asList("deep", "muted", "bright", "pastel", "dark", "colorblind");

            public String setTheme(String style, String context, String palette) {
                if (!valid_styles.contains(style)) {
                    style = "darkgrid";
                }
                if (!valid_contexts.contains(context)) {
                    context = "notebook";
                }
                if (!valid_palettes.contains(palette)) {
                    palette = "deep";
                }

                
                StandardChartTheme theme = new StandardChartTheme("Seaborn-like");
                if (style.equals("dark") || style.equals("darkgrid")) {
                    theme.setPlotBackgroundPaint(Color.DARK_GRAY);
                }

                return String.format("{'style': '%s', 'context': '%s', 'palette': '%s'}", style, context, palette);
            }
        }

        SeabornThemeSetter themeSetter = new SeabornThemeSetter();
        return themeSetter.setTheme(style, context, palette);
    }
}

File: data/python/src/function_seaborn_set_theme.py
import seaborn as sns

class SeabornThemeSetter:
    def set_theme(self, style="darkgrid", context="notebook", palette="deep") -> str:
        
        valid_styles = ["white", "dark", "whitegrid", "darkgrid", "ticks"]
        valid_contexts = ["paper", "notebook", "talk", "poster"]
        valid_palettes = ["deep", "muted", "bright", "pastel", "dark", "colorblind"]

        if style not in valid_styles:
            style = "darkgrid"  
        if context not in valid_contexts:
            context = "notebook" 
        if palette not in valid_palettes:
            palette = "deep"
        
        sns.set_theme(style=style, context=context, palette=palette)
        return f"{{'style': '{style}', 'context': '{context}', 'palette': '{palette}'}}"
